File: Week6/pset6/mario/mario.py
from cs50 import get_int

def main():
    height = get_int("Height: ")
    while (height <=0 or type(height) != int or height > 8):
        print("Invalid Number.")
        height = get_int("Height: ")
    pyramid_height = height+1
    for i in range(1, pyramid_height):
        for row in range(1,pyramid_height-i):
            print(" ", end="")
        for row in range(i):
            print("#", end="")
            if row == i-1:
                print("")
if __name__ == "__main__":

    main()

File: Week1/Pset1/mario.c
#include <stdio.h>
#include <cs50.h>
void print_pyramid(int n, int h);
int pyramid_height(void);
int main (void)
{
    //ask the user the height of the pyramid;
    int height = pyramid_height();
    //build the pyramid
    for (int i = 1; i <= height; i++)
    {
        print_pyramid(i, height);
    }
}
//This funcion print a row of the pyramid. row 1 get  # #, row 2 gets two ## ##
void print_pyramid(int n, int h)
{
    for(int j=0; j<(h-n); j++)
    {
        printf(" ");
    }
    for (int i=0; i< n; i++)
    {

        printf ("#");
    }
    printf (" ");
    for (int i=0; i< n; i++)
    {
        printf ("#");
    }
    printf("\n");
}

// Function that asks for the height. The height muss to be between 0 and 8;

int pyramid_height(void)
{
    int height;
    do
    {
    height = get_int("Height: ");
    } while (height > 8 || height < 1);
    return height;
}